backend/test_observer/controllers/test_cases/test_cases.py

```diff
@@ -29,7 +30,20 @@
 
 
 @router.get("", response_model=TestCasesResponse)
-def get_test_cases(db: Session = Depends(get_db)) -> TestCasesResponse:
+def get_test_cases(
+    q: Annotated[
+        str | None,
+        Query(description="Search term for test case names"),
+    ] = None,
+    limit: Annotated[
+        int | None,
+        Query(ge=1, le=1000, description="Maximum number of results (optional)"),
```

Comment:
I think you should set a default limit. It's a bit odd to support between 1 and 1000, or unlimited.